docs/2_3_common_states.adoc

```diff
@@ -82,485 +87,644 @@ If `loggingOn == fmi3True`, the FMU enables a vendor defined set of `<LogCategor
++
+[[fmi3UpdateDiscreteStates,`fmi3UpdateDiscreteStates`]]
 [source, C]
 ----
-include::../headers/fmi3FunctionTypes.h[tags=EnterContinuousTimeMode]
+include::../headers/fmi3FunctionTypes.h[tag=UpdateDiscreteStates]
 ----
 
-TODO only in ME
+[[discreteStatesNeedUpdate,`discreteStatesNeedUpdate`]]
+* When `discreteStatesNeedUpdate == fmi3True`, the importer must stay in <<EventMode>> for another event iteration.
 
-The model enters *Continuous-Time Mode* and all discrete-time equations become inactive and all relations are "frozen". +
-This function has to be called when changing from *Event Mode* (after the global event iteration in *Event Mode* over all involved FMUs and other models has converged) into *Continuous-Time Mode*. +
+[[terminateSimulationUDS,`terminateSimulation`]]
+* When <<terminateSimulationUDS,`terminateSimulation == fmi3True`>>, the FMU requests to stop the simulation and the importer must call <<fmi3Terminate>>.
 
-_[This function might be used additionally for the following purposes:_
-* _If the FMU stores results internally on file, then the results after the initialization and/or the event has been processed can be stored._
-* _If the FMU contains dynamically changing states, then a new state selection might be performed with this function.]_
+[[nominalsOfContinuousStatesChanged,`nominalsOfContinuousStatesChanged`]]
+* If argument `nominalsOfContinuousStatesChanged == fmi3True`, then at least one nominal value of the <<state,`states`>> has changed and can be inquired with <<fmi3GetNominalsOfContinuousStates>>. +
+This argument is only valid in Model Exchange.
 
-Function <<fmi3EnterStepMode>>::
+[[valuesOfContinuousStatesChanged,`valuesOfContinuousStatesChanged`]]
+* If argument `valuesOfContinuousStatesChanged == fmi3True`, then at least one continuous state has changed its value. +
+[[state,`state`]]
+The new values of the <<state,`states`>> can be inquired with <<fmi3GetContinuousStates>> or individually for each state for which <<reinit, `reinit = true`>> by calling `fmi3GetFloat64`. +
+This argument is only valid in Model Exchange.
 
-TODO only in CS
+[[nextEventTimeDefined,`nextEventTimeDefined`]]
+* When <<nextEventTimeDefined,`nextEventTimeDefined == fmi3True`>>, the next time event is known and <<nextEventTime>> has a valid value.
 
-Once all events are handled and `discreteStatesNeedUpdate == fmi3False`, the FMU should be pushed to *Step Mode* by calling <<fmi3EnterStepMode>>, unless it requests to terminate the Co-Simulation by setting  <<terminateSimulation,`terminateSimulation`> to `fmi3True`.
-In this case, a new step can be started from the current communication point time.
+[[nextEventTime,`nextEventTime`]]
+* The return argument <<nextEventTime>> contains the absolute time of the next time event latexmath:[\mathbf{T}_\mathit{next}] if <<nextEventTimeDefined,`nextEventTimeDefined == fmi3True`>>.
```

Comment:
For time-based aperiodic input clocks with changing interval: How is the next interval defined. with the argement Nexst EventTime of fmi3UpdateDiscrete states? Or by a fmi3GetInterval call?

Reply:
We tried to make this clear in the table listing the clocks: for aperiodic clocks, it is GetInterval and depending on the kind the importer has to call it in every Event Mode or only in those where the clock ticked.

nextEventTime argument of UpdateDiscreteStates is used for time events only. I think this is what we have written there.

Reply:
> nextEventTime argument of UpdateDiscreteStates is used for time events only.

So we use nextEventTime only for time events that are not associated with a time-based clock?

Reply:
Yes. @TorstenBlochwitz - please confirm.

Reply:
Then I would explicitely state this, something like: 
For time events associated to time-based clocks, fmi3GetInterval shall be used to to to communicate the next event time instead of  nextEventTimeDefined and  nextEventTime.

Reply:
No, sorry I did not make this clear: Time events don´t have to be associated with a clock. Time events can be purely internal and still trigger event handling. We will have to find a central place where to describe this. Please open another issue, I want to finally merge this PR.

Reply:
Reading 2.3.2.1, I now understand it; no PR needed from my side.